File: frontend/src/services/ai.ts
// src/services/ai.ts
import api from './api'; // 1. 引入封装好的 api 实例，而不是原生 axios

export interface ChatMessage {
  role: 'user' | 'assistant';
  content: string;
  // _system:string
}

export const sendChatMessage = async (messages: ChatMessage[]) => {
  // 2. 使用 api.post
  // api.ts 的 baseURL 是 '/api'
  // ai.py 的路由是 prefix='/ai' + '/chat'
  // 最终请求 URL 会自动拼接为: /api/ai/chat
  const response = await api.post('/ai/chat', { messages }, {
    timeout: 60000
  });
  return response.data;
};

File: backend/app/routers/ai.py
# backend/app/routers/ai.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import List
import logging
from app.services.ai_service import get_travel_advice, generate_diary_draft

# 获取 logger
logger = logging.getLogger(__name__)

router = APIRouter(prefix="/ai", tags=["AI Assistant"])

class Message(BaseModel):
  role: str
  content: str

class ChatRequest(BaseModel):
  messages: List[Message]

@router.post("/chat")
async def chat_with_ai(request: ChatRequest):
  logger.info(f"收到 AI 聊天请求，消息条数: {len(request.messages)}")

  try:
    # 限制上下文长度
    if len(request.messages) > 20:
      request.messages = request.messages[-20:]

    # 调用 Service
    response_text = await get_travel_advice([m.model_dump() for m in request.messages])

    logger.info("AI 响应生成完毕")
    return {"role": "assistant", "content": response_text}

  except Exception as e:
    logger.error(f"路由层处理 AI 请求失败: {str(e)}", exc_info=True)
    raise HTTPException(status_code=500, detail="AI Service Error")


class GenerateDiaryRequest(BaseModel):
  prompt: str
@router.post("/generate-diary")
async def generate_diary(request: GenerateDiaryRequest):
  logger.info(f"收到生成日记请求: {request.prompt[:50]}...")

  try:
    data = await generate_diary_draft(request.prompt)

    if data is None:
      raise HTTPException(status_code=500, detail="Failed to generate valid JSON from AI")

    return data
  except Exception as e:
    logger.error(f"生成日记路由错误: {str(e)}", exc_info=True)
    raise HTTPException(status_code=500, detail=str(e))